relay/relay.go

```diff
@@ -383,6 +389,16 @@ func (rs *Relay) GetPayload(ctx context.Context, m *structs.MetricGroup, payload
 		"processingTimeMs": time.Since(tStart).Milliseconds(),
 	})
 
+	if rs.exp != nil {
+		go func() {
```

Comment:
Can we create deferred goroutine that would run both SubmitGetPayloadRequest and PutDelivered based on the flags? This way we'll have half of them running, and we don't need this to be saved immeditely :)

Reply:
No, we cannot, really, because `SubmitGetPayloadRequest` occurs even if the request is not successful, while `PutDelivered` only when it's successful

Reply:
You can gather flags from both actions so like are we erroring or not. and run that goroutine inside defer, so that you can - inside goroutine - skip PutDelivered if that wasn't